spring-batch-infrastructure/src/main/java/org/springframework/batch/support/transaction/TransactionAwareBufferedWriter.java

```diff
@@ -197,8 +239,30 @@ public void close() throws IOException {
 	@Override
 	public void flush() throws IOException {
 		if (!transactionActive() && forceSync) {
-			channel.force(false);
+			outputStream.getChannel().force(false);
+		}
+	}
+	
+	@Override
+	public void write(String str, int off, int len) throws IOException {
+		if (!transactionActive()) {
+			if (len > 8192) {
```

Comment:
The code tries to be really clever but by looking at `OutputStreamWriter#write` it will eventually end up calling `sun.nio.cs.StreamEncoder#write(String, int, int)` which will end up allocating a `char[len]` so it won't actually end up saving anything. We would have to iterate for that.
I'll simply remove the code.